amazon/ion/simpleion.py

```diff
@@ -489,8 +496,19 @@ def dump_extension(obj, fp, binary=True, sequence_as_stream=False, tuple_as_sexp
     fp.write(res)
 
 
-def load_extension(fp, single_value=True, parse_eagerly=True):
-    iterator = ionc.ionc_read(fp, emit_bare_values=False)
+def load_extension(fp, single_value=True, parse_eagerly=True, symbol_buffer_threshold=None):
+    """
+    Args:
+        fp (str): A string representation of Ion data.
+        single_value (Optional[True|False]): When True, the data in ``ion_str`` is interpreted as a single Ion value,
+            and will be returned without an enclosing container. If True and there are multiple top-level values in
+            the Ion stream, IonException will be raised. NOTE: this means that when data is dumped using
+            ``sequence_as_stream=True``, it must be loaded using ``single_value=False``. Default: True.
+        parse_eagerly: (Optional[True|False]) Used in conjunction with ``single_value=False`` to return the result as list
+            or an iterator
+        symbol_buffer_threshold: The size maximum size allowed for symbols, 512 bytes is the default
```

Comment:
Two comments:
1) I think we should use a different name than `symbol_buffer_threshold`. Does this buffer limit the size of both symbols and strings? If so, something like `text_buffer_size_limit` would be more accurate.
2) The public documentation for this flag should be explicit about the fact that when the C extension is enabled (and it is enabled by default), raising this limit is required in order to read text values that exceed 512 bytes.

Reply:
Below two commits addressed above comments perspectively
1. Yes, changed to `text_buffer_size_limit`
2. Added.